### PR TITLE
BGP speaker 4 byte ASN

### PR DESCRIPTION
**Pull request by vaniramdurg**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
test_bgp_speaker.py is the motivation for this PR
#### How did you do it?
Added blocks that can change DUT and/or bgp speaker configuration to configure 4byte ASN.
#### How did you verify/test it?
I ran the ./run_test command and tested exabgp session establishment, route announce, and traffic send and receive.
#### Any platform specific information?
Currently tested with t0 and PTF.
#### Supported testbed topology if it's a new test case?
Sonic t0 with ceos neighbors.

### CODE REVIEW

File: tests/bgp/test_bgp_speaker_2b4b.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
+"""
+This test is for verifying bgp session establishing, route announcing and traffic flow 
+from a 2 byte ASN DUT to 4 byte ASN bgp speakers.
+The script is similar to test_bgp_speaker.py, except that bgp speaker ASN modification.
```
Comment (lguohan) [inline tests/bgp/test_bgp_speaker_2b4b.py:4]: please reuse the test_bgp_speaker.py test, do not create a new test. we need to reuse our code. having duplicated code increase the cost of maintainance

**Review by lguohan — CHANGES_REQUESTED** [review]
please reuse the test_bgp_speaker.py test, do not create a new test. we need to reuse our code. having duplicated code increase the cost of maintainance

File: docs/testplan/BGP-Speaker-2byte-to-4byte-ASN.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Testbed
+The test is developed and tested with Sonic t0 testbed setup and CEOS neighbors.
+
+## Setup configuration
+This test doesn't required any configurations from outside. But intenally it will modify the ASN of BGPVac and BGPSLBPassive groups in which the new neighboars are being created dynamically. Ensure the "bgp_speaker_route_4bASN.j2" file is available under sonic-mgmt/tests/bgp/templates.
+
+## Test
+Configure testbed to emulate production.
+Configure IXIA/Spirent with a 4-byte ASN and peer with DUT.
```
Comment (lguohan) [inline docs/testplan/BGP-Speaker-2byte-to-4byte-ASN.md:20]: why ixia here? are we using ixia here? if not, please remove from the description.

**Comment by vaniramdurg** [comment]
Created another one based on review comments. Merged these changes inside test_bgp_speaker.py

**Pull request closed without merging**